### PR TITLE
Skydive-Prometheus connector blog

### PR DESCRIPTION
**Pull request by KalmanMeth**
Prometheus (`https://prometheus.io/`) is an open-source systems monitoring and alerting tool.
Many different types of data are collected by different tools and are forwarded to Prometheus via various types of exporters.
We developed an exporter for Skydive, based on the skydive-flow-exporter, that reports metrics of individual captured network flows.
The connector translates data from Skydive captured flows into a format that can be consumed by Prometheus.
The first implementation of the Skydive-Prometheus connector periodically provides the byte transfer counts for each network connection under observation.
The code can be easily tailored to provide additional flow information.

### CODE REVIEW

**Comment by KalmanMeth** [comment]
@eranra @eran.raichstein@gmail.com @hunchback Please review. Thanks.

**Comment by eranra** [comment]
@KalmanMeth good :-) 

Can you add list of metrics that are exposed and the labels .... ?

**Comment by eranra** [comment]
@KalmanMeth also if you can also add an example of deployment and /o r configuration that will help someone that wants to use that to be successful in this deployment

Look here on "testing" (and maybe other sections) as reference:: https://www.syslog-ng.com/community/b/blog/posts/prometheus-syslog-ng-exporter

**Comment by KalmanMeth** [comment]
> 
> 
> @KalmanMeth good :-)
> 
> Can you add list of metrics that are exposed and the labels .... ?

The first implementation of the Skydive-Prometheus connector periodically provides the byte transfer counts for each network connection under observation. This variable and the labels are specified in the section "Verify Operation".

**Comment by KalmanMeth** [comment]
> @KalmanMeth also if you can also add an example of deployment and /o r configuration that will help someone that wants to use that to be successful in this deployment
> 
An example of deployment appears towards the end of the page under "Prometheus setup" and "Verify Operation".

**Comment by eranra** [comment]
> > @KalmanMeth good :-)
> > Can you add list of metrics that are exposed and the labels .... ?
> 
> The first implementation of the Skydive-Prometheus connector periodically provides the byte transfer counts for each network connection under observation. This variable and the labels are specified in the section "Verify Operation".

@KalmanMeth  If you can add that to the text, maybe in a form of a table with columns such as of metric name, description,  and label ... it will help and build for the future.

**Comment by eranra** [comment]
@KalmanMeth I looked again on the blog ... maybe another small thing that might be nice to show is output of prometheus ... text or graph or screenshot ... showing how things look like in prometheus

File: blog/prometheus-connector.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  * `initiator_ip`
+  * `target_ip`
+  * `initiator_port`
+  * `target_port`
+  * `direction`
+  * `node_tid`
+
+In addition, each such tuple is associated with timestamps and corresponding values that appear in the graph.
+
+##
```
Comment (safchain) [inline blog/prometheus-connector.md]: Is there any missing content here ? It seems to be a title

**Review by safchain — COMMENTED** [review]
Thanks for submitting this blog post. I think it would be nice to add link(s) to the exporter repository.

I add to make some changes to have jekyll generating it properly.

https://paste.centos.org/view/204e53b2

File: blog/prometheus-connector.md (hunk 1 above)
Comment (KalmanMeth) [reply blog/prometheus-connector.md]: Addressed reviewers comments. Pushed commits with updates.

**Comment by KalmanMeth** [comment]
> @KalmanMeth If you can add that to the text, maybe in a form of a table with columns such as of metric name, description, and label ... it will help and build for the future.

Done.

**Comment by KalmanMeth** [comment]
> @KalmanMeth I looked again on the blog ... maybe another small thing that might be nice to show is output of prometheus ... text or graph or screenshot ... showing how things look like in prometheus

Added a screenshot of a graph from prometheus.

File: blog/prometheus-connector.md (hunk 1 above)
Comment (KalmanMeth) [reply blog/prometheus-connector.md]: Extra heading was removed.
Other edit suggestions incorporated in update.

File: blog/prometheus-connector.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  </a>
+</p>
+
+## Deploy Skydive
+
+You can either download a statically linked version of Skydive or you can build it on your own.
+
+To download a pre-compiled statically linked version of Skydive, run the following command.
+
++{% highlight shell %}
```
Comment (safchain) [inline blog/prometheus-connector.md]: There are some extra `+` remaining in the file that need to be removed. It seems to be only before jekyll tags.

Comment (KalmanMeth) [reply blog/prometheus-connector.md]: Fixed.

**Comment by safchain** [comment]
Thanks for the blog post.

**Pull request merged**